common/models/user.js

```diff
@@ -969,6 +1067,47 @@ module.exports = function(User) {
     next();
   });
 
+  User.observe('before save', function rejectInsecurePasswordChange(ctx, next) {
+    const UserModel = ctx.Model;
+    if (UserModel.settings.legacyPasswordFlow !== false) {
+      // In legacyPasswordFlow, any DAO method can change the password
+      return next();
+    }
+
+    if (ctx.isNewInstance) {
+      // The password can be always set when creating a new User instance
+      return next();
+    }
+    const isPasswordChange = 'password' in (ctx.data || ctx.instance);
```

Comment:
What do you mean by string search? Could you paste a code snippet showing what do you have in mind?